src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Options/GraphQLContentOptions.cs

```diff
@@ -221,5 +222,17 @@ public bool IsHiddenByDefault(ContentTypePartDefinition definition)
 
             return false;
         }
+
+        internal string GetFieldName(ContentTypePartDefinition definition, string partName, string fieldName)
+        {
+            var settings = definition.GetSettings<GraphQLContentTypePartSettings>();
+
+            if (settings.PreventFieldNameCollision)
+            {
+                return $"{partName.ToFieldName()}{fieldName.ToPascalCase()}";
```

Comment:
No separators?

Reply:
@sebastienros yes, no separator, **partName.ToFieldName()** is producing **camelCase**, **fieldName.ToPascalCase()** is producing **PascalCase** when concatenated you will get some user friendly fields

example:
partName: MyCustomPart
fieldName: MyCustomField
will produce: **myCustomMyCustomField** graphQL field

![image](https://github.com/OrchardCMS/OrchardCore/assets/6403130/fb6bddf3-e14b-40fe-bb98-4a4e3dcd3f9e)